EIPS/eip-7525.md

```diff
@@ -0,0 +1,161 @@
+
+The cost that varies with the length of the input consists of 1 **`RLP decoding`** and 1 **`verifyProof`** operation, both proportional to the size of the input. These operations incur a cost of 3 Gwei per word.
+
+The total recovery cost is as follows:
+
+$$
+\text{Total Restoration Cost} = 37000 + 900\times{Epoch} + 3\times{words} + \text{Memory Cost}
+$$
+
+## Rationale
+
+The rationale behind this EIP is anchored on enhancing network efficiency, scalability, and accessibility to the network.
+
+### Reduction of Storage Requirements
+
+By expiring inactive accounts, we can reduce the storage requirements for nodes significantly.  This makes the network more accessible for participants and improving decentralization.
+
+### Node Modes
+
+There will be two modes for nodes: archive mode, which stores all data, and compact mode, which retains data only for active accounts. Archive mode operates the same as before, while the compact mode node can validate the transaction data of the active accounts it maintains. The compact mode node is sufficient for most situations. If historical data is needed, it can be recovered with the help of archive nodes.
+
+### Ensuring Data Accessibility
+
+It's important to note that this EIP does not propose discarding any data. Instead, data related to inactive accounts will be moved to archive nodes and can be accessed when necessary.
+
+For the restoration of historical data, it is necessary to maintain a sufficient number of nodes operating in archive mode. To encourage this, we need to establish incentives for nodes to run in archive mode. Nodes operating in this mode would earn fees in exchange for supplying the data required for restoration. These fees would be determined based on principle of supply and demand.
+
+In addition to the approach suggested here, within the Ethereum community, there are active discussions on various methodologies aimed at ensuring data accessibility during the history and state expiration processes.
+
+### Enhanced Network Efficiency
+
+By focusing on a smaller set of active accounts, the overall efficiency of the network improves. This enhancement can also potentially speed up block validation times and reduce propagation times, leading to better network performance.
+
+## Backwards Compatibility
+
+With the implementation of this EIP, a new field is added to the account state and this field is used in the calculation of the nonce. This means that the nonce can increase by a value other than 1 during the sweep or restoration process. In other words, the Tx Pool should no longer assume that the nonce increases by 1.
+
+Since the nonce alone cannot sufficiently fulfill its role, it is safe to use the `restored_epoch` together with the nonce in other parts where the nonce was used.
```

Comment:
why not just restore nonce too? whats the reasoning for starting from zero

Reply:
Actually, it includes nonce restoration. Please refer to the link below :)
https://github.com/ethereum/EIPs/pull/7776/files#diff-0cd84964790721de12937d83aa6be4002a46f170da91bb89f55f33c457d0fbe6R90